packages/wp-now/src/pool.ts

```diff
@@ -0,0 +1,261 @@
+import { NodePHP } from '@php-wasm/node';
+
+let childCount = 0;
+
+/**
+ * PRIVATE
+ * Tracks stats of instances in a pool.
+ */
+class PoolInfo {
+	id = childCount++; // Unique ID for debugging purposes.
+	requests = 0; // Total requests processed.
+	started = Date.now(); // Time spawned.
+	active = false; // Whether instance is considered active.
+}
+
+/**
+ * PRIVATE
+ * Spawns new instances if the pool is not full.
+ * Returns a list of new instances.
+ */
+const spawn = (pool) => {
+	const newInstances = new Set();
+
+	while (pool.maxJobs > 0 && pool.instanceInfo.size < pool.maxJobs) {
+		const info = new PoolInfo();
+		const instance = pool.spawner();
```

Comment:
What do you think about `spawn()` as the name here instead of `spawner()`?

Reply:
@adamziel 
Originally, this function (const spawn = (pool) => {...}) WAS `Pool.spawn`, until I made it private and remove it from the object, so I had to use a different name for the provided callback.

Now that I've refactored it into a private function, its completely possible for me to use the correct, verb based name for the function.